src/lib.rs

```diff
@@ -123,46 +126,80 @@ extern crate pest;
 /// and thus the using can be shortened to the following one:
 /// # Examples:
 /// ```
+/// use std::str::FromStr;
 /// use serde_json::{json,Value};
-/// use jsonpath_rust::JsonPathQuery;
+/// use crate::jsonpath_rust::{JsonPathFinder,JsonPathQuery,JsonPathInst};
 ///fn test(){
-///         let json: Value = serde_json::from_str("{}").expect("to get json");
-///         let v = json.path("$..book[?(@.author size 10)].title").expect("the path is correct");
+///         let json: Value = serde_json::from_str("{}").unwrap();
+///         let v = json.path("$..book[?(@.author size 10)].title").unwrap();
 ///         assert_eq!(v, json!([]));
-/// }
+///
+///         let json: Value = serde_json::from_str("{}").unwrap();
+///         let path = &json.path("$..book[?(@.author size 10)].title").unwrap();
+///
+///         assert_eq!(path, &json!(["Sayings of the Century"]));
+///
+///         let json: Box<Value> = serde_json::from_str("{}").unwrap();
+///         let path: Box<JsonPathInst> = Box::from(JsonPathInst::from_str("$..book[?(@.author size 10)].title").unwrap());
+///         let finder = JsonPathFinder::new(json, path);
+///
+///         let v = finder.find_slice();
+///         assert_eq!(v, vec![&json!("Sayings of the Century")]);
+///     }
+///
 /// ```
 /// #Note:
 /// the result is going to be cloned and therefore it can be significant for the huge queries
 pub trait JsonPathQuery {
     fn path(self, query: &str) -> Result<Value, String>;
 }
 
+pub struct JsonPathInst {
+    inner: JsonPath,
+}
+
+
+impl FromStr for JsonPathInst{
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        JsonPath::from_str(s).map(|inner| JsonPathInst { inner })
+    }
+}
+
+impl JsonPathQuery for Box<Value> {
+    fn path(self, query: &str) -> Result<Value, String> {
+        let p = JsonPathInst::from_str(query)?;
+        Ok(JsonPathFinder::new(self, Box::new(p)).find())
+    }
+}
+
 impl JsonPathQuery for Value {
     fn path(self, query: &str) -> Result<Value, String> {
-        let p = parse_json_path(query).map_err(|e| e.to_string())?;
-        Ok(JsonPathFinder::new(self, p).find())
+        let p = JsonPathInst::from_str(query)?;
+        Ok(JsonPathFinder::new(Box::from(self), Box::new(p)).find())
     }
 }
 
 
-/// The base structure conjuncting the json instance and jsonpath instance
+/// The base structure stitching the json instance and jsonpath instance
 pub struct JsonPathFinder {
-    json: Value,
-    path: JsonPath,
+    json: Box<Value>,
```

Comment:
Here the `JsonPathFinder` still owned the `serde_json::Value`, and worse it limits the json value must be boxed(heap).